src/estimagic/optimization/tranquilo/solve_subproblem.py

```diff
@@ -85,20 +86,7 @@ def get_subsolver(solver, user_options=None, bounds=None):
             "or a callable."
         )
 
-    default_options = {
-        "maxiter": 20,
-        "maxiter_gradient_descent": 5,
-        "conjugate_gradient_method": "cg",
-        "gtol_abs": 1e-8,
-        "gtol_rel": 1e-8,
-        "gtol_scaled": 0,
-        "gtol_abs_conjugate_gradient": 1e-8,
-        "gtol_rel_conjugate_gradient": 1e-6,
-        "k_easy": 0.1,
-        "k_hard": 0.2,
-    }
-
-    all_options = {**default_options, **user_options}
+    all_options = SubsolverOptions()._replace(**user_options)._asdict()
```

Comment:
Also use `update_option_bundle` to get type conversion

Reply:
This will yield a circular import. This would, however, be fixed in the upcoming subsolver PR, when we adapt the subsolver code to the `get_component` framework.